proxy/tests/discovery.rs

```diff
@@ -32,6 +34,77 @@ macro_rules! generate_tests {
             assert_eq!(client.get("/recon"), "nect");
         }
 
+        #[test]
+        #[cfg_attr(not(feature = "flaky_tests"), ignore)]
+        fn outbound_destinations_reset_on_reconnect_followed_by_no_endpoints_exists() {
+            outbound_destinations_reset_on_reconnect(move || {
+                Some(controller::destination_exists_with_no_endpoints())
+            })
+        }
+
+        #[test]
+        #[cfg_attr(not(feature = "flaky_tests"), ignore)]
+        fn outbound_destinations_reset_on_reconnect_followed_by_add_none() {
+            outbound_destinations_reset_on_reconnect(move || {
+                Some(controller::destination_add_none())
+            })
+        }
+
+        #[test]
+        #[cfg_attr(not(feature = "flaky_tests"), ignore)]
+        fn outbound_destinations_reset_on_reconnect_followed_by_remove_none() {
+            outbound_destinations_reset_on_reconnect(move || {
+                Some(controller::destination_remove_none())
+            })
+        }
+
+        fn outbound_destinations_reset_on_reconnect<F>(f: F)
```

Comment:
Does this need a `#[test]` attribute? And I guess a `#[cfg_attr(not(feature = "flaky_tests"), ignore)]`, due to the thread sleep flakiness?

Reply:
> Does this need a #[test] attribute? 

No. `outbound_destinations_reset_on_reconnect()` isn't a test itself. It is a helpful function used by the three tests that are marked `#[test]`.

>  And I guess a #[cfg_attr(not(feature = "flaky_tests"), ignore)], due to the thread sleep flakiness?

Yes, sadly. I'm expecting it will get resolved after we have mock timers or some other mechanism.